WordPress/gutenberg
# feat: Add Hook to monitor network connectivity status

## Related
- https://github.com/wordpress-mobile/gutenberg-mobile/pull/6417
- https://github.com/wordpress-mobile/WordPress-iOS/pull/22112
- https://github.com/wordpress-mobile/WordPress-Android/pull/19692

## What?

Add a React Hook for monitoring the network connection status via the bridge to
the host app.

## Why?

The network connection status will enable future features improving the UX when
a user's network connection changes, e.g. displaying status indicators.
Fixes https://github.com/wordpress-mobile/gutenberg-mobile/issues/6409. 

## How?

Add a `useIsConnected` React Hook that invokes bridge methods retrieving and
monitoring the network connection status from the host app. Relying upon the
host app will result in a single source of truth and mitigate conflicting
statuses.

## Testing Instructions




1. Checkout this branch within the `gutenberg-mobile` submodule. 
2. Start the Metro server via `npm run start:reset`. 
3. Apply the following patch: 
    <details><summary>Test Diff</summary>

    ```diff
    diff --git a/packages/block-editor/src/components/block-list/index.native.js b/packages/block-editor/src/components/block-list/index.native.js
    index 810e23e4c1..9ab21320cb 100644
    --- a/packages/block-editor/src/components/block-list/index.native.js
    +++ b/packages/block-editor/src/components/block-list/index.native.js
    @@ -1,7 +1,7 @@
    /**
     * External dependencies
     */
    -import { View, Platform, Pressable } from 'react-native';
    +import { View, Platform, Pressable, Text } from 'react-native';
    
    /**
     * WordPress dependencies
    @@ -15,6 +15,7 @@ import {
        alignmentHelpers,
    } from '@wordpress/components';
    import { __ } from '@wordpress/i18n';
    +import { useIsConnected } from '@wordpress/react-native-bridge';
    
    /**
     * Internal dependencies
    @@ -227,6 +228,7 @@ export default function BlockList( {
            headerToolbar.height +
            blockToolbar.height +
            ( isFloatingToolbarVisible ? floatingToolbar.height : 0 );
    +	const { isConnected } = useIsConnected();
    
        return (
            <View
    @@ -235,6 +237,20 @@ export default function BlockList( {
                onLayout={ onLayout }
                testID="block-list-wrapper"
            >
    +			<View
    +				style={ {
    +					backgroundColor: 'red',
    +				} }
    +			>
    +				<Text
    +					style={ {
    +						color: 'white',
    +						fontSize: 30,
    +					} }
    +				>
    +					Connected: { isConnected === null ? 'null' : isConnected }
    +				</Text>
    +			</View>
                { isRootList ? (
                    <BlockListProvider
                        value={ {

    ```

    </details>
4. Configure the app to load the local Metro server:
    - For iOS, install pods via `LOCAL_GUTENBERG=true bundle exec pod install`. 
    - For Android, set `localGutenbergMobilePath` within `local-builds.gradle`. 
7. Build and run the WordPress app. 
8. Launch the editor and verify the connection status displayed is accurate. 
9. Toggle airplane mode, verify the connection status displayed is accurate. 

### Testing Instructions for Keyboard

n/a, this internal tool introduces no user-facing changes.

## Screenshots or screencast 
n/a, this internal tool introduces no visible changes.
